bigquery/schema.go

```diff
@@ -533,6 +542,11 @@ func inferFieldSchema(fieldName string, rt reflect.Type, nullable bool) (*FieldS
 		return &FieldSchema{Required: !nullable, Type: BooleanFieldType}, nil
 	case reflect.Float32, reflect.Float64:
 		return &FieldSchema{Required: !nullable, Type: FloatFieldType}, nil
+	case reflect.Map:
+		if rt.Key().Kind() != reflect.String {
+			return nil, unsupportedFieldTypeError{fieldName, rt}
```

Comment:
I wonder if this is a case where we should either augment `unsupportedFieldTypeError` or provide a more specific error to help users identify that the issue is the map key rather than the map type itself.  Generally I'm okay as is, but may be something worth revisiting if we get feedback from users confused by the error.

Reply:
Yeah that's reasonable. As it is the error will say `type map[int]interface {} is not supported` at least, which should help more than just saying `type int is not supported` or something like that.